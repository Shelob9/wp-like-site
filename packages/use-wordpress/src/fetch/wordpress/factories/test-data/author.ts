export default {
  id: 1,
  name: 'Josh Pollock',
  url: 'https://joshpress.net',
  description:
    'Josh Pollock is VP of Engineering Experience at <a href="https://saturdaydrive.io/">Saturday Drive</a>. Previously, he was the founder of CalderaWP and the lead developer of Caldera Forms. He is also a WordPress core contributor and the author of two books about WordPress development.',
  link: 'https://calderaforms.com/author/josh/',
  slug: 'josh',
  avatar_urls: {
    '24':
      'https://secure.gravatar.com/avatar/90f0d6d84cc52508796d039eb81c4f06?s=24&d=mm&r=g',
    '48':
      'https://secure.gravatar.com/avatar/90f0d6d84cc52508796d039eb81c4f06?s=48&d=mm&r=g',
    '96':
      'https://secure.gravatar.com/avatar/90f0d6d84cc52508796d039eb81c4f06?s=96&d=mm&r=g',
  },
  meta: [],
  _links: {
    self: [{ href: 'https://calderaforms.com/wp-json/wp/v2/users/1' }],
    collection: [{ href: 'https://calderaforms.com/wp-json/wp/v2/users' }],
  },
};
