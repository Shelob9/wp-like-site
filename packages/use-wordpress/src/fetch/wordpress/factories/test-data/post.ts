export default {
  id: 168826,
  date: '2019-12-17T11:16:33',
  date_gmt: '2019-12-17T16:16:33',
  guid: { rendered: 'https://calderaforms.com/?p=168826' },
  modified: '2019-12-17T11:51:49',
  modified_gmt: '2019-12-17T16:51:49',
  slug: '2019-holiday-update',
  status: 'publish',
  type: 'post',
  link: 'https://calderaforms.com/2019/12/2019-holiday-update/',
  title: { rendered: 'A Quick Holiday Update!' },
  content: {
    rendered: '\n<h2>Bug fixes as a present are kind of like getting',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Bug fixes as a present are kind of like getting socks for Christmas&#8230; not the most thrilling thing, but when you need socks&#8230; The holidays are often a different kind of time for teams. We want to keep moving all the things forward, but it&#8217;s also healthy to slow down and appreciate family, too. The &hellip; <a href="https://calderaforms.com/2019/12/2019-holiday-update/" class="more-link">Continue reading <span class="screen-reader-text">A Quick Holiday Update!</span></a></p>\n',
    protected: false,
  },
  author: 62578,
  featured_media: 168887,
  comment_status: 'open',
  ping_status: 'open',
  sticky: false,
  template: '',
  format: 'standard',
  meta: [],
  categories: [174],
  tags: [],
  _links: {
    self: [{ href: 'https://calderaforms.com/wp-json/wp/v2/posts/168826' }],
    collection: [{ href: 'https://calderaforms.com/wp-json/wp/v2/posts' }],
    about: [{ href: 'https://calderaforms.com/wp-json/wp/v2/types/post' }],
    author: [
      {
        embeddable: true,
        href: 'https://calderaforms.com/wp-json/wp/v2/users/62578',
      },
    ],
    replies: [
      {
        embeddable: true,
        href: 'https://calderaforms.com/wp-json/wp/v2/comments?post=168826',
      },
    ],
    'version-history': [
      {
        count: 5,
        href: 'https://calderaforms.com/wp-json/wp/v2/posts/168826/revisions',
      },
    ],
    'predecessor-version': [
      {
        id: 168925,
        href:
          'https://calderaforms.com/wp-json/wp/v2/posts/168826/revisions/168925',
      },
    ],
    'wp:featuredmedia': [
      {
        embeddable: true,
        href: 'https://calderaforms.com/wp-json/wp/v2/media/168887',
      },
    ],
    'wp:attachment': [
      { href: 'https://calderaforms.com/wp-json/wp/v2/media?parent=168826' },
    ],
    'wp:term': [
      {
        taxonomy: 'category',
        embeddable: true,
        href: 'https://calderaforms.com/wp-json/wp/v2/categories?post=168826',
      },
      {
        taxonomy: 'post_tag',
        embeddable: true,
        href: 'https://calderaforms.com/wp-json/wp/v2/tags?post=168826',
      },
      {
        taxonomy: 'caldera_admin_help',
        embeddable: true,
        href:
          'https://calderaforms.com/wp-json/wp/v2/caldera_admin_help?post=168826',
      },
      {
        taxonomy: 'caldera_doc_overview',
        embeddable: true,
        href:
          'https://calderaforms.com/wp-json/wp/v2/caldera_doc_overview?post=168826',
      },
    ],
    curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }],
  },
};
