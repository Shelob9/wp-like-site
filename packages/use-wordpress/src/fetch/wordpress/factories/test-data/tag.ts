export default {
  id: 515,
  count: 1,
  description: '',
  link: 'https://calderaforms.com/tag/hiroy/',
  name: '#hiRoy',
  slug: 'hiroy',
  taxonomy: 'post_tag',
  _links: {},
};
