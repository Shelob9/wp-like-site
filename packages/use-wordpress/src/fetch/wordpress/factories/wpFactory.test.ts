import factory, {
  authorFromWpApi,
  featuredFromWpApi,
  tagFromWpApi,
  tagsFromWpApi,
} from './wpFactory';
import author from './test-data/author';
import mediaItem from './test-data/media';
import tag from './test-data/tag';
import post from './test-data/post';
test('convert wp author', () => {
  const converted = authorFromWpApi(author);
  expect(converted.name).toEqual('Josh Pollock');
  expect(converted.link?.href).toEqual('https://calderaforms.com/author/josh/');
});

test('convert featured', () => {
  const converted = featuredFromWpApi(mediaItem);
  expect(converted.src).toEqual(mediaItem.source_url);
  expect(converted.width).toEqual(2508);
  expect(converted.height).toEqual(1673);
});

test('convert tag', () => {
  //@ts-ignore
  const converted = tagFromWpApi(tag);
  expect(converted.slug).toEqual(tag.slug);
  expect(converted.label).toEqual(tag.name);
});

test('convert tags', () => {
  //@ts-ignore
  const converted = tagsFromWpApi([tag]);
  expect(converted[0].slug).toEqual(tag.slug);
  expect(converted[0].label).toEqual(tag.name);
});

test('factory', () => {
  const getters = {
    author: () => author,
    featured: () => mediaItem,
    tags: () => {
      //@ts-ignore
      return [tag];
    },
    published: () => post.date,
  };
  //@ts-ignore
  const f = factory(getters);
  const data = f.convertPost(post);
  expect(data).toMatchSnapshot();
});

test('factory deals with featured image being 0', () => {
  const fail = true;
  const getters = {
    author: () => author,
    featured: () => {
      return new Promise((resolve, reject) => {
        if (fail) {
          reject();
        }
        resolve();
      });
    },
    tags: () => {
      //@ts-ignore
      return [tag];
    },
    published: () => post.date,
  };
  //@ts-ignore
  const f = factory(getters);
  const data = f.convertPost(post);
  expect(data).toMatchSnapshot();
});
