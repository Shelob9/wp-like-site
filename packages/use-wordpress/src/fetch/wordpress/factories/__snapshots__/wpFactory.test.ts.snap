// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factory 1`] = `
Object {
  "author": Object {
    "avatar": Object {
      "alt": "Avatar for Josh Pollock",
      "src": "https://secure.gravatar.com/avatar/90f0d6d84cc52508796d039eb81c4f06?s=96&d=mm&r=g",
    },
    "description": "Josh Pollock is VP of Engineering Experience at <a href=\\"https://saturdaydrive.io/\\">Saturday Drive</a>. Previously, he was the founder of CalderaWP and the lead developer of Caldera Forms. He is also a WordPress core contributor and the author of two books about WordPress development.",
    "id": 1,
    "link": Object {
      "href": "https://calderaforms.com/author/josh/",
      "text": "Josh Pollock",
    },
    "name": "Josh Pollock",
    "slug": "josh",
  },
  "content": Object {
    "protected": false,
    "rendered": "
<h2>Bug fixes as a present are kind of like getting",
  },
  "excerpt": Object {
    "protected": false,
    "rendered": "<p>Bug fixes as a present are kind of like getting socks for Christmas&#8230; not the most thrilling thing, but when you need socks&#8230; The holidays are often a different kind of time for teams. We want to keep moving all the things forward, but it&#8217;s also healthy to slow down and appreciate family, too. The &hellip; <a href=\\"https://calderaforms.com/2019/12/2019-holiday-update/\\" class=\\"more-link\\">Continue reading <span class=\\"screen-reader-text\\">A Quick Holiday Update!</span></a></p>
",
  },
  "featured": Object {
    "alt": "CF Show Pro UI",
    "height": 1673,
    "src": "https://calderaforms.com/wp-content/uploads/2019/12/StockSnap_61AE320141.jpg",
    "width": 2508,
  },
  "id": "168826",
  "published": "2019-12-17T11:16:33",
  "slug": "2019-holiday-update",
  "tags": Array [
    Object {
      "label": "#hiRoy",
      "slug": "hiroy",
    },
  ],
  "title": Object {
    "rendered": "A Quick Holiday Update!",
  },
}
`;

exports[`factory deals with featured image being 0 1`] = `
Object {
  "author": Object {
    "avatar": Object {
      "alt": "Avatar for Josh Pollock",
      "src": "https://secure.gravatar.com/avatar/90f0d6d84cc52508796d039eb81c4f06?s=96&d=mm&r=g",
    },
    "description": "Josh Pollock is VP of Engineering Experience at <a href=\\"https://saturdaydrive.io/\\">Saturday Drive</a>. Previously, he was the founder of CalderaWP and the lead developer of Caldera Forms. He is also a WordPress core contributor and the author of two books about WordPress development.",
    "id": 1,
    "link": Object {
      "href": "https://calderaforms.com/author/josh/",
      "text": "Josh Pollock",
    },
    "name": "Josh Pollock",
    "slug": "josh",
  },
  "content": Object {
    "protected": false,
    "rendered": "
<h2>Bug fixes as a present are kind of like getting",
  },
  "excerpt": Object {
    "protected": false,
    "rendered": "<p>Bug fixes as a present are kind of like getting socks for Christmas&#8230; not the most thrilling thing, but when you need socks&#8230; The holidays are often a different kind of time for teams. We want to keep moving all the things forward, but it&#8217;s also healthy to slow down and appreciate family, too. The &hellip; <a href=\\"https://calderaforms.com/2019/12/2019-holiday-update/\\" class=\\"more-link\\">Continue reading <span class=\\"screen-reader-text\\">A Quick Holiday Update!</span></a></p>
",
  },
  "featured": Object {
    "alt": undefined,
    "height": undefined,
    "src": undefined,
    "width": undefined,
  },
  "id": "168826",
  "published": "2019-12-17T11:16:33",
  "slug": "2019-holiday-update",
  "tags": Array [
    Object {
      "label": "#hiRoy",
      "slug": "hiroy",
    },
  ],
  "title": Object {
    "rendered": "A Quick Holiday Update!",
  },
}
`;
